RELEASE/scripts/autoscend/combat/auto_combat_ed.ash

```diff
@@ -118,58 +115,58 @@ string auto_edCombatHandler(int round, monster enemy, string text)
 		}
 	}
 
+	//iotm back item and the enemies it spawns (free fights) can be killed using special skills to get extra XP and item drops
 	if(have_equipped($item[Protonic Accelerator Pack]) && isGhost(enemy))
 	{
-		if(canUse($skill[Summon Love Gnats]))
+		skill stunner = getStunner(enemy);
+		if(stunner != $skill[none])
 		{
-			return useSkill($skill[Summon Love Gnats]);
+			return useSkill(stunner);
 		}
 
-		if(auto_have_skill($skill[Shoot Ghost]) && (my_mp() > mp_cost($skill[Shoot Ghost])) && !contains_text(edCombatState, "shootghost3") && !contains_text(edCombatState, "trapghost"))
+		if(canUse($skill[Shoot Ghost], false) && !combat_status_check("shootghost3"))
 		{
 			boolean shootGhost = true;
-			if(contains_text(edCombatState, "shootghost2"))
+			if(combat_status_check("shootghost2"))
 			{
 				if((damageReceived * 1.075) > my_hp())
 				{
 					shootGhost = false;
 				}
 				else
 				{
-					set_property("auto_edCombatHandler", edCombatState + "(shootghost3)");
+					combat_status_add("shootghost3");
 				}
 			}
-			else if(contains_text(edCombatState, "shootghost1"))
+			else if(combat_status_check("shootghost1"))
 			{
 				if((damageReceived * 2.05) > my_hp())
 				{
 					shootGhost = false;
 				}
 				else
 				{
-					set_property("auto_edCombatHandler", edCombatState + "(shootghost2)");
+					combat_status_add("shootghost2");
 				}
 			}
 			else
 			{
-				set_property("auto_edCombatHandler", edCombatState + "(shootghost1)");
+				combat_status_add("shootghost1");
 			}
 
 			if(shootGhost)
 			{
-				return "skill " + $skill[Shoot Ghost];
+				return useSkill($skill[Shoot Ghost], false);
 			}
 			else
 			{
-				edCombatState += "(trapghost)(love stinkbug)";
-				set_property("auto_edCombatHandler", edCombatState);
+				combat_status_add("trapghost");
```

Comment:
Similar to above comment. Why add trapghost to combat status if we don't check for it later

Reply:
It was initially necessary for the following function that happens afterwards in combat. But that function was refactored into modern code as well making it obsolete. So Now that 
```
else
			{
				combat_status_add("trapghost");
			}
```
can and should be deleted.

in fact I think I should further refactor the entire ghostbusting function. one moment

Reply:
refactored the entire proton pack combat code further